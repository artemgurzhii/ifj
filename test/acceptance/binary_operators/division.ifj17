Scope
dim a as integer
dim b as float
dim c as integer

a = 5
b = 2.2

c = a / b

End Scope
