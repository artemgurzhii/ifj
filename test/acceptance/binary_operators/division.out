.IFJcode17
JUMP Scope
LABEL Scope
CREATEFRAME
DEFVAR TF@temp_bool_1
DEFVAR TF@temp_bool_2
DEFVAR GF@a
DEFVAR GF@b
DEFVAR GF@c
MOVE GF@a int@5
MOVE GF@b float@2.2
TYPE TF@temp_bool_1 GF@a
TYPE TF@temp_bool_2 float@1.5
JUMPIFNEQ RES_IF_int2float_1 TF@temp_bool_1 TF@temp_bool_2
JUMP CONT_1
LABEL RES_IF_int2float_1
INT2FLOAT GF@a GF@a
LABEL CONT_1
TYPE TF@temp_bool_1 GF@b
TYPE TF@temp_bool_2 float@1.5
JUMPIFNEQ RES_IF_int2float_2 TF@temp_bool_1 TF@temp_bool_2
JUMP CONT_2
LABEL RES_IF_int2float_2
INT2FLOAT GF@b GF@b
LABEL CONT_2
DIV GF@c GF@a GF@b
LABEL END_IF_0
